src/pdfnorm.jl

```diff
@@ -0,0 +1,50 @@
+"""
+    pdfsquaredL2norm(d::Distribution)
+
+Return the square of the L2 norm of the probability density function ``f(x)`` of the distribution `d`:
+
+```math
+\\int_{S} f(x)^{2} \\mathrm{d} x
+```
+
+where ``S`` is the support of ``f(x)``.
+"""
+pdfsquaredL2norm
+
+function pdfsquaredL2norm(d::Beta)
+    α, β = params(d)
+    z = beta(2 * α - 1, 2 * β - 1) / beta(α, β) ^ 2
+    # L2 norm of the pdf converges only for α > 0.5 and β > 0.5
+    return α > 0.5 && β > 0.5 ? z : oftype(z, Inf)
+end
+
+pdfsquaredL2norm(d::Cauchy) = inv(twoπ * d.σ)
```

Comment:
A micro-optimization would be
```suggestion
pdfsquaredL2norm(d::Cauchy) = inv2π / d.σ
```